packages/core/src/metadata/MetadataDiscovery.ts

```diff
@@ -441,9 +441,14 @@ export class MetadataDiscovery {
 
       if (value instanceof Function) {
         const meta2 = this.metadata.get(prop.type);
-        prop[type] = value(meta2.properties).name;
+        
```

Comment:
isn't this the exact same, but with literally "1 char addition" instead of 5 LoC diff? (note the optional chanining)
```suggestion
        prop[type] = value(meta2.properties)?.name;
```